### PR TITLE
2020-09-22 GUARD-833 Handle-Huge-Product-Pages

### PR DESCRIPTION
**Pull request by fabulaspb**
**Summary**
Some customers have products with tons of variations (~600), so JSON response can have huge size, more than 10MB (usually over 100KB).
BigCommerce server immediately closes connection after passing over 4MB. Found out this when tried to read incoming data through buffers of certain size.
But BigCommerce server supports response compression and I used this feature to solve our issue.

### CODE REVIEW

**Review by maxim-skuvault — APPROVED** [review]

File: src/BigCommerceAccess/Services/WebRequestServices.cs [hunk 1]
```diff
@@ -235,41 +236,23 @@ private HttpWebRequest CreateServicePutRequest( string url, string content )
 		private BigCommerceResponse< T > ParseResponse< T >( WebResponse response, string marker ) where T : class
 		{
 			using( var stream = response.GetResponseStream() )
-			using( var reader = new StreamReader( stream ) )
 			{
-				var jsonResponse = reader.ReadToEnd();
+				using ( var reader = new StreamReader( stream ) )
```
Comment (slav) [inline src/BigCommerceAccess/Services/WebRequestServices.cs:240]: Why did you change formatting here?

Comment (fabulaspb) [reply src/BigCommerceAccess/Services/WebRequestServices.cs:240]: @slav Added curly braces around each using statement to make code clear. Do we have some formatting rules about this?

**Review by slav — APPROVED** [review]
